WirescaleApi/WireguardManager.cs

```diff
@@ -0,0 +1,55 @@
+﻿using NetTools;
+
+namespace WirescaleApi;
+
+public class WireguardManager
+{
+    private readonly IWgrestApiClient _wgrestApiClient;
+    private readonly IpAddressAllocator _ipAddressAllocator;
+
+    public WireguardManager(IWgrestApiClient wgrestApiClient, IpAddressAllocator ipAddressAllocator)
+    {
+        _wgrestApiClient = wgrestApiClient;
+        _ipAddressAllocator = ipAddressAllocator;
+    }
+
+    public async Task<WireguardPeerRegistrationResult> RegisterNewPeer(string clientPublicKey)
+    {
+        var deviceName = "wg-wirescale";
+        var wireguardDevices = await _wgrestApiClient.GetWgrestDevices();
+        var wirescaleDevice = wireguardDevices.Single(x => x.Name == deviceName);
+
+        var wireguardPeers = await _wgrestApiClient.GetWgrestPeers(wirescaleDevice.Name);
+        if (wireguardPeers.Any(x => x.PublicKey == clientPublicKey))
+        {
+            throw new InvalidOperationException("Public key is already registered as a peer");
+        }
+
+        // We currently only support one (IPv4) network. Throw an exception if device has more or less than that.
+        var wireguardNetworkCidr = wirescaleDevice.Networks.Single();
+
+        // Get the currently registered peer IP ranges
+        var currentPeerIpRanges = wireguardPeers.Select(x => IPAddressRange.Parse(x.AllowedIps.Single()));
+
+        var clientPeerIpAddress = _ipAddressAllocator.AllocateIpAddress(wireguardNetworkCidr, currentPeerIpRanges);
```

Comment:
If I understand correctly, this can be called concurrently if multiple peers are being registered simultaneously, correct? Wouldn't it result in a possibility of allocating the same IP address for multiple peers then, since current peer ranges are passed in here?

Reply:
Pushed a small change to address this issue https://github.com/runesoerensen/wirescale/pull/4/commits/7f3faba0e869953ef9519587cec14d2066a468a2